### PR TITLE
INDEXING-270

### PR DESCRIPTION
**Pull request by deelawn**
Fixed bug preventing task finishing.  There is one thing I wonder about -- why not just do something like:
```
c.runLock.Lock()
c.run[s.ID] = s
c.runLock.Unlock()
```
And then replace references to `r` with references to `s`.  Is it insane?

### CODE REVIEW

File: core/core.go [hunk 1]
```diff
@@ -109,10 +109,19 @@ func (c *Core) LoadScheduler(ctx context.Context) error {
 		r, ok := c.run[s.ID]
 		if !ok {
 			c.run[s.ID] = s
```
[the remark this answers is not in the document]
Comment (tasinco) [reply core/core.go:111]: Possible fix:
```
r=s
c.run[s.ID] = r
```
?

But there are cleaner fixes..

```
if !ok {
  r = s
}
r.Status = s.Status
c.run[s.ID] = r
```

Then r is set and updated to the correct status?

[the remark this answers is not in the document]
Comment (deelawn) [reply core/core.go:111]: Everything you wrote makes sense, but as a bit of a more detailed explanation --

So the current changes are strictly an attempt to fix the reported bug.  This is why I included an additional suggestion in the PR description regarding something that might be more correct; it seems like we'd want to just overwrite whatever is in memory with the most recent value returned from the database. I don't think it matters if it is creating a new entry in the map or overwriting an old one. Does that make sense?

File: core/core.go [hunk 2]
```diff
@@ -109,10 +109,19 @@ func (c *Core) LoadScheduler(ctx context.Context) error {
...
+			c.run[s.ID] = r
 		}
 		c.runLock.Unlock()
 
-		if r.Enabled && (r.Status != structures.StateFinished && r.Status != structures.StateStopped) {
+		shouldEnable := r.Status != structures.StateFinished &&
+			r.Status != structures.StateStopped &&
+			r.Status != structures.StateRunning
+
+		if r.Enabled && shouldEnable {
```
Comment (lukanus) [inline core/core.go:124]: I don't feel we should assign just to test :) you can make it multiline here if you like it to be

Comment (deelawn) [reply core/core.go:124]: It also can't be in the running state. There is an edge case where it may be in the running state when the database is queried, but updated to finished state before the check that is in `c.EnableSchedule`.  So it all comes back to the question in the PR description -- can we just do that and change `r` to `s` as you demonstrated in your example above?

File: core/core.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tasinco) [reply core/core.go:111]: The code does have an issue when `r` is not initialized.  I think it should be fixed.

[the remark this answers is not in the document]
Comment (deelawn) [reply core/core.go]: I just applied the changes from my question in the PR description. Sorry for the confusion. What do you think?

**Review by tasinco — APPROVED** [review]
